src/traces/box/defaults.js

```diff
@@ -43,6 +43,9 @@ function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
     coerce('boxmean', boxmeanDflt);
 
     coerce('whiskerwidth');
+    coerce('showwhiskers');
+    coerce('sizemode');
```

Comment:
```suggestion
    var sizemode = coerce('sizemode');
    coerce('showwhiskers', sizemode === 'quartiles');
```
if I'm understanding correctly that most users of `sd` mode would not want to see whiskers? And then if we do this, we'd also want to take [`dflt`](https://github.com/plotly/plotly.js/pull/6698/files#diff-591038d6258ff010e19e85399d7314ab4024713b2b5892ef8bcf443307145cf4R406) out of the `showwhiskers` attribute definition and describe this logic in its `description`.

Reply:
Oh also with `sizemode='sd'` the `boxmean` attribute just doesn't make sense, right? We'd just be drawing a diamond that matches the box. So let's move the `coerce('boxmean', boxmeanDflt);` from above down here and put it inside an `if(sizemode === 'quartiles')`